src/training_strategies.jl

```diff
@@ -296,3 +296,80 @@ function get_loss_function(loss_function, lb, ub, eltypeθ, strategy::Quadrature
     loss = (θ) -> 1 / area * f_(lb, ub, loss_function, θ)
     return loss
 end
+
+struct WeightedGridTraining <: AbstractTrainingStrategy
+    dx::Float64
+    weights::Vector{Float64}
+    samples::Int
+end
+
+function WeightedGridTraining(dx, weights)
+    WeightedGridTraining(dx, weights)
+end
+
+function merge_strategy_with_loss_function(pinnrep::PINNRepresentation,
+                                           strategy::WeightedGridTraining,
+                                           datafree_pde_loss_function,
+                                           datafree_bc_loss_function)
+    @unpack domains, eqs, bcs, dict_indvars, dict_depvars, flat_init_params = pinnrep
+    dx = strategy.dx
+    eltypeθ = eltype(pinnrep.flat_init_params)
+
+    train_sets = generate_training_sets(domains, dx, eqs, bcs, eltypeθ,
+                                        dict_indvars, dict_depvars)
+
+    # the points in the domain and on the boundary
+    pde_train_sets, bcs_train_sets = train_sets
+
+    pde_train_sets = adapt.(typeof(flat_init_params), pde_train_sets)
+    bcs_train_sets = adapt.(typeof(flat_init_params), bcs_train_sets)
+    pde_loss_functions = [get_loss_function(_loss, _set, eltypeθ, strategy)
+                          for (_loss, _set) in zip(datafree_pde_loss_function,
+                                                   pde_train_sets)]
+
+    bc_loss_functions = [get_loss_function(_loss, _set, eltypeθ, strategy)
+                         for (_loss, _set) in zip(datafree_bc_loss_function, bcs_train_sets)]
+
+    pde_loss_functions, bc_loss_functions
+end
```

Comment:
this isn't called?

Reply:
Yep, I just created it in case someone else wants to use it with the new strategy. I can remove it if you'd like

Reply:
add a test that uses it with PDEs then?